packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts

```diff
@@ -89,8 +89,12 @@ const getOrganizedParticipants = (props: OrganizedParticipantsArgs): OrganizedPa
 
   let gridParticipants = newGridParticipants;
   let overflowGalleryParticipants = newOverflowGalleryParticipants;
-  if (gridParticipants.length + callingParticipants.length <= maxGridParticipants) {
-    gridParticipants = gridParticipants.concat(callingParticipants);
+  if (overflowGalleryParticipants.length === 0) {
+    const numberOfCallingParticipantsInGrid = maxGridParticipants - gridParticipants.length;
+    const gridCallingParticipants = callingParticipants.slice(0, numberOfCallingParticipantsInGrid);
+    const overflowGalleryCallingParticipants = callingParticipants.slice(numberOfCallingParticipantsInGrid);
+    gridParticipants = gridParticipants.concat(gridCallingParticipants);
```

Comment:
It's really hard to parse when reading the difference between `participantsForGrid`, `newGridParticipants`, `gridParticipants` and `gridCallingParticipants`, is there anything we can do to improve this chunk?

Reply:
Perhaps even just a comment like:
```
// Add the participants being called into the call.
// If there are already overflow participants, add these to the array of overflow participants
// Otherwise, add the maximum number to the main grid, then put the rest in the overflow
```